www/include/TestResults.php

```diff
@@ -186,8 +186,17 @@ public function getMedianRunNumber($metric, $cached, $medianMode = "median") {
     if ($numValues == 1 || $medianMode == "fastest") {
       return $runNumbers[0];
     }
-    $medianIndex = (int)floor($numValues / 2.0);
-    return $runNumbers[$medianIndex];
+    $medianIndex = (int)floor(((float)$numValues + 1.0) / 2.0);
```

Comment:
You are right, there should be only one. The new function supports for multistep data though, and the old one doesn't. Multistep needed some big changes and is not yet supported in all parts of WPT, so it wasn't possible to stick with only one method.

Nevertheless, I think using a loop instead of using a `-1` is adding confusing to the code, which needs to be different to the original anyway.